src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_ev_driver_posix.cc

```diff
@@ -98,12 +99,71 @@ class GrpcPolledFdPosix : public GrpcPolledFd {
 
 class GrpcPolledFdFactoryPosix : public GrpcPolledFdFactory {
  public:
+  ~GrpcPolledFdFactoryPosix() {
```

Comment:
Seems like the `GrpcPolledFdFactoryPosix` object needs to outlive the [ares_destroy](https://github.com/grpc/grpc/blob/master/src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper.cc#L210) call? Is it guaranteed?

Reply:
Yes that's guaranteed because:
- `GrpcPolledFdFactoryPosix` is [owned](https://github.com/grpc/grpc/blob/498fc99479a01177aa2ad257f1390521495b0788/src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper.cc#L1290 by the request object
- the request object is only deleted after the query completes
- the query only completes after all `GrpcPolledFd` object have been destroyed